### PR TITLE
Fixing issues in the design view

### PR DESCRIPTION
**Pull request by NatashaWso2**
Fixes: https://github.com/ballerinalang/composer/issues/2471
Fixes: https://github.com/ballerinalang/composer/issues/2313
Fixes: https://github.com/ballerinalang/composer/issues/2614
Fixes: https://github.com/ballerinalang/composer/issues/2611

### CODE REVIEW

File: modules/web/js/ballerina/ast/ballerina-ast-root.js [hunk 1]
```diff
@@ -176,12 +176,28 @@ class BallerinaASTRoot extends ASTNode {
     }
 
     /**
+     * Adds an import from a json object
+     * @param {Object} jsonNode json object representing the import
+     * @returns {void}
```
Comment (NipunaMarcus) [inline modules/web/js/ballerina/ast/ballerina-ast-root.js]: this might give lint errors saying return description not available.

Comment (NatashaWso2) [inline modules/web/js/ballerina/ast/ballerina-ast-root.js]: Fixed with commit a4cb957

File: modules/web/js/ballerina/ast/ballerina-ast-root.js [hunk 2]
```diff
@@ -176,12 +176,27 @@ class BallerinaASTRoot extends ASTNode {
...
+            return;
+        }
+        importNode.initFromJson(jsonNode);
+        this.addImport(importNode);
+    }
+
+
+    /**
      * Adds new import declaration.
      * @param {ImportDeclaration} importDeclaration - New import declaration.
```
Comment (NipunaMarcus) [inline modules/web/js/ballerina/ast/ballerina-ast-root.js]: identifier name should be changed from importDeclaration to importNode

File: modules/web/js/ballerina/ast/import-declaration.js [hunk 3]
```diff
@@ -117,6 +117,11 @@ class ImportDeclaration extends ASTNode {
      * @param jsonNode
      */
     initFromJson(jsonNode) {
+        if (!_.isNil(jsonNode.whitespace_descriptor)) {
+            this.whiteSpace.currentDescriptor = jsonNode.whitespace_descriptor;
+            this.whiteSpace.useDefault = false;
```
Comment (NipunaMarcus) [inline modules/web/js/ballerina/ast/import-declaration.js]: are we activating useDefault later for this?

File: modules/web/js/ballerina/components/package-definition.jsx [hunk 4]
```diff
@@ -142,10 +142,18 @@ class PackageDefinition extends React.Component {
...
      */
     handleAddImport(value) {
-        const newImportDeclaration = BallerinaASTFactory.createImportDeclaration();
-        newImportDeclaration.setPackageName(value);
-        newImportDeclaration.setParent(this.props.model.parent, { doSilently: true });
-        this.props.model.parent.addImport(newImportDeclaration);
+        if (!value) {
+            return;
+        }
+        value = 'import ' + value + ';\n';
```
Comment (NipunaMarcus) [inline modules/web/js/ballerina/components/package-definition.jsx]: Just a suggestion. can we add the '\n' as a whitespace region if we have white spaces for this?

File: modules/web/js/ballerina/ast/ballerina-ast-root.js (hunk 2 above)
Comment (NatashaWso2) [inline modules/web/js/ballerina/ast/ballerina-ast-root.js]: Changed the identifier name as importDeclaration to be consistent in commit 3680b229404cacc4c299e52adc1692db2c581a16

File: modules/web/js/ballerina/ast/import-declaration.js (hunk 3 above)
Comment (NatashaWso2) [inline modules/web/js/ballerina/ast/import-declaration.js]: It is activated in node.js

File: modules/web/js/dialog/new-item-dialog.js [hunk 5]
```diff
@@ -31,6 +31,10 @@ class NewItemDialog extends ModalDialog {
 
     onSubmit(data, itemName) {
         const app = this._options.application;
+
+        if (!itemName.includes('.bal')) {
```
Comment (hemikak) [inline modules/web/js/dialog/new-item-dialog.js]: Can use https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith as well. Also, lowercase the itemName and check. Have to consider '.BAL' too.

File: modules/web/js/ballerina/item-provider/initial-definitions.js [hunk 6]
```diff
@@ -279,12 +279,11 @@ const createTransactionAbortedStatementTool = {
...
-    createVariableDefinitionStatementTool, createFunctionInvocationTool, createReturnStatementTool,
-    createReplyStatementTool, createWhileStatementTool, createBreakStatementTool, createTryCatchStatementTool,
-    createThrowStatementTool, createWorkerInvocationStatementTool, createWorkerReplyStatementTool,
-    createTransformStatementTool, createJoinStatementTool, createAbortStatementTool,
-    createTransactionAbortedStatementTool, createContinueStatementTool, createNamespaceDeclarationStatementTool];
+const statementToolDefArray = [createAbortStatementTool, createAssignmentExpressionTool, createBreakStatementTool,
+    createContinueStatementTool, createJoinStatementTool, createFunctionInvocationTool, createIfStatementTool,
+    createNamespaceDeclarationStatementTool, createWorkerReplyStatementTool, createReplyStatementTool,
+    createReturnStatementTool, createWorkerInvocationStatementTool, createThrowStatementTool, createTransactionAbortedStatementTool,
+    createTransformStatementTool, createTryCatchStatementTool, createVariableDefinitionStatementTool, createWhileStatementTool];
```
Comment (hemikak) [inline modules/web/js/ballerina/item-provider/initial-definitions.js]: Does this line goes beyond 120 chars  ?

File: modules/web/js/ballerina/components/package-definition.jsx [hunk 7]
```diff
@@ -142,10 +142,18 @@ class PackageDefinition extends React.Component {
...
+        parseContent(value)
+            .then((jsonTree) => {
+                if (!jsonTree.root[1]) {
```
Comment (hemikak) [inline modules/web/js/ballerina/components/package-definition.jsx]: Wont it make sense if we have if (jsonTree.root[1]) { 

We can avoid the return statement as well.

Comment (NatashaWso2) [inline modules/web/js/ballerina/components/package-definition.jsx]: Fixed with commit 8776f92

File: modules/web/js/ballerina/item-provider/initial-definitions.js (hunk 6 above)
Comment (NatashaWso2) [inline modules/web/js/ballerina/item-provider/initial-definitions.js]: Fixed with commit 8776f92

File: modules/web/js/dialog/new-item-dialog.js (hunk 5 above)
Comment (NatashaWso2) [inline modules/web/js/dialog/new-item-dialog.js]: Both the lowercase and uppercase extensions were considered.
 Fixed with commit 8287007

**Pull request merged**